code/backtracking/src/sudoku_solve/sudoku_solveNxN.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include<bits/stdc++.h>
+
+using namespace std;
+
+const int n = 9;
+
+int grid[n][n];
+
+void input_grid() {
+    int i, j;
+    for (i = 0; i < n; i++) {
```

Comment:
Use pre-increment instead of post-increment. Also declare loop variables inside the loop itself. For example:

```suggestion
    for (int i = 0; i < n; ++i) {
```

Reply:
It's doesn't make an effect declaring again again we can declare it once it's easy

Reply:
@aryangulati But it's not following _best practices_. Sure it may not have a major effect on this program specifically, but when someone comes along and see this, they may think its normal and use it in their own code. Reusing loop variables is _never_ a good idea. Each loop should have its own code, especially in C++.

Reply:
I agree with all but not this because no one ever take my loop and use it whosoever know basic would be able to understand the logic behind is SAME

Reply:
I disagree, because it's definitely clearer when you can immediately see where a variable comes from, but that's your opinion.